proto-public/pbmesh/v2beta1/mesh_gateway.proto

```diff
@@ -5,4 +5,8 @@ syntax = "proto3";
 
 package hashicorp.consul.mesh.v2beta1;
 
-message MeshGateway {}
+import "pbresource/annotations.proto";
+
+message MeshGateway {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_CLUSTER};
```

Comment:
I think this should be `SCOPE_PARTITION` because you can have mutliple mesh gateways in a cluster potentially living in different partitions

Reply:
Good catch! Addressed